assets/app/features/upgrade/controllers/upgrade-database-configuration.controller.spec.js

```diff
@@ -64,14 +94,124 @@ describe('Upgrade Flow - Create Connect Database Controller', function () {
 
     it('should display "Password" field', function () {});
 
+    describe('createServer function', function () {
+
+        it('is defined', function() {
+            should.exist(controller.createServer);
+        });
+
+        it('should do nothing if fields are empty', function () {
+            spyOn(upgradeFactory, 'createNewDatabaseServer');
+            controller.createServer();
+            $rootScope.$digest();
+            expect(upgradeFactory.createNewDatabaseServer).not.toHaveBeenCalled()
+        });
+
+        describe('when success', function () {
+            beforeEach(function () {
+                bard.mockService(upgradeFactory, {
+                    createNewDatabaseServer: $q.when(successResponse)
+                });
+
+                controller.databaseForm = {
+                    username: 'test',
+                    password: 'test'
+                };
+
+                controller.createServer();
+                $rootScope.$digest();
+            });
+
+            it('errors should be empty', function () {
+                expect(controller.errors).toEqual([]);
```

Comment:
@santiph this will not work as jasmine is overriding chai's `expect()`. To make this work you would need to locally switch to chai e.g. `var expect = window.chai.expect;` or call it directly via `chai.expect(...).to.be.empty;`.